Scripts/modelsystem.py

```diff
@@ -373,49 +346,76 @@ def _calculate_noise_areas(self):
         data["population"] = conversion * data["area"] * pop
         self.resultdata.print_data(data, "noise_areas.txt")
 
-    def _calculate_accessibility_and_savu_zones(self):
-        zone_numbers = self.zdata_forecast.zone_numbers
-        access = {modes: pandas.Series(0, zone_numbers, name="all")
-            for modes in ("total", "car", "sustainable")}
+    def _export_accessibility(self):
         for purpose in self.dm.tour_purposes:
-            if purpose.name in gen_param.tour_weights:
-                zone_numbers = purpose.zone_numbers
-                bounds = purpose.bounds
-                weight = gen_param.tour_weights[purpose.name]
-                for modes in access:
-                    mode_access = purpose.accessibility_model.access[modes]
-                    access[modes][bounds] += weight * mode_access
-        pop = self.zdata_forecast["population"][bounds]
-        for modes in access:
-            self.resultdata.print_data(
-                access[modes], f"{modes}_accessibility.txt")
-            avg_access = numpy.average(access[modes][bounds], weights=pop)
-            self.resultdata.print_line(
-                f"\n{modes.title()} accessibility:\t{avg_access:1.2f}",
-                "result_summary")
-            if modes == "sustainable":
-                intervals = zone_param.savu_intervals
-                savu = numpy.searchsorted(intervals, access[modes][bounds]) + 1
+            accessibility = purpose.accessibility_model.accessibility
+            for key in accessibility:
+                logsum = pandas.Series(numpy.log(accessibility[key]), 
+                    purpose.zone_numbers, name=f"{purpose.name}_{key}")
+                self.resultdata.print_data(logsum, f"accessibility.txt")
+    
+    def _export_resultdata(self):
```

Comment:
I think it is a good idea to have all the result printing here. I think a good compromise (considering oop) and a nice solution for making this method cleaner would be to have several simple `@property` getter methods in `TourPurpose`. For instance:

```
@property
def all_generated_tours(self):
    return pandas.Series(
        sum(self.generated_tours.values()), self.zone_numbers, name=self.name)
```

The same goes for `modes_shares`, `trip_lengths`, etc.

Reply:
Updated this and I'll re-request review.